2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -133,13 +139,22 @@ class CollectorService(
         `Access-Control-Allow-Headers`("Content-Type")
       ))
 
+  override def flashCrossDomainPolicy: HttpResponse =
+    flashCrossDomainPolicy(config.crossDomain)
+
   /** Creates a response with a cross domain policiy file */
-  override def flashCrossDomainPolicy: HttpResponse = HttpResponse(
-    entity = HttpEntity(
-      contentType = ContentType(MediaTypes.`text/xml`, HttpCharsets.`ISO-8859-1`),
-      string = "<?xml version=\"1.0\"?>\n<cross-domain-policy>\n  <allow-access-from domain=\"*\" secure=\"false\" />\n</cross-domain-policy>"
-    )
-  )
+  def flashCrossDomainPolicy(crossDomainConfig: Option[CrossDomainConfig]): HttpResponse =
+    crossDomainConfig
+      .map { c =>
+        HttpResponse(
+          entity = HttpEntity(
+            contentType = ContentType(MediaTypes.`text/xml`, HttpCharsets.`ISO-8859-1`),
+            string = s"""<?xml version=\"1.0\"?>\n<cross-domain-policy>
+                        |  <allow-access-from domain=\"${c.domain}\" secure=\"${c.secure}\" />
+                        |</cross-domain-policy>""".stripMargin
+          )
+        )
+      }.getOrElse(HttpResponse(404, entity = "404 not found"))
```

Comment:
I think pattern-matching would be more readable here.

Reply:
good point 👍